classes/helper/HelperCalendar.php

```diff
@@ -148,7 +148,7 @@ public function getDateFormat()
     public function setDateFrom($value = '')
     {
         if (empty($value)) {
-            $value = strtotime('-31 days');
+            $value = '-31 days';
         }
```

Comment:
Why not like this?
```suggestion
    public function setDateFrom($value = '-31 days')
    {
```

Reply:
Yes, this update is possible

Reply:
I'm not sure about this modification, it's not the same thing The previous implementation was resilient to passing an empty string as the parameter (the cause of the bug as I understand) whereas the new one only allows to call `setDateFrom` without a parameter, but it this parameter is empty the bug is still present
So in my opinion your initial fix as better

Reply:
I see, if $value = '' on call method, this variable not initialized with default values in params ?
I test recently my second commit on a website, on dashboard this fix is functional but on other call I don't know.
Is it possible to had other review on second commit ?

On php.net (my first commit is better) : 
Use of default parameters in functions

<?php
function makecoffee($type = "cappuccino")
{
    return "Making a cup of $type.\n";
}
echo makecoffee();
echo makecoffee(null);
echo makecoffee("espresso");
?>
The above example will output:
Making a cup of cappuccino.
Making a cup of .
Making a cup of espresso.

I remove last commit ?

Reply:
I think the point is to get back to the first proposal, where you check whether the value is empty. @jolelievre seems to be correct.